theurgy.lic

```diff
@@ -40,11 +47,84 @@ def comm_tasmine
   put_in_container($settings.water_holder)
 end
 
+def comm_kert
+  return unless DRStats.circle > 8 && DRSkill.getxp('Theurgy') <= 31
+  case DRC.bput("get oil from #{$settings.theurgy_supply_container}", 'holy oil', 'some oil', 'I could not find', 'What were you referring to')
+  when 'I could not find', 'What were you referring to'
+    return
+  when 'holy oil'
+  when 'some oil'
+    if 'interrupt your research' == DRC.bput('prep bless', 'Bless spell', 'interrupt your research')
+      research_type = ["Fundamental", "Augmentation", "Stream", "Sorcery", "Utility", "Warding"]
+      find_current_project = DRC.bput("research status", / (\d+)% complete with a portion .*\./)
+      hold_research = research_type.find{ |hold_research| find_current_project =~ /#{hold_research}/i }
+      if hold_research == nil
+        return
+      end
+      if find_current_project[1,2].to_i >= 60
+        wait_time = (101 - find_current_project[1,2].to_i) * 3
+        pause (wait_time)
+      else
+        echo "stopping research"
+      end
+      fput 'prep bless'
+    end	
+    pause 5
+    fput 'cast oil'
+  end
+  fput "sprinkle holy oil on #{checkname}"
+  put_in_container('oil')
+  fput "commune kert"
+end
+
+def comm_truff
+  return unless DRSkill.getxp('Theurgy') <= 24 && DRStats.circle > 60
+  fput "stow #{GameObj.right_hand.noun}" if GameObj.right_hand.name != 'Empty'
+  fput "stow #{GameObj.left_hand.noun}" if GameObj.left_hand.name != 'Empty'
+  get_from_container('Taffelberries')
+  fput 'commune truff'
+  pause 1
+  if checkright == 'orb'
+    fput 'drop orb'
+  else
+    fput 'get Taffelberries'
+    put_in_container('Taffelberries')
+  end
+end
+
+######################## SHOPPING ########################
+
+def check_supplies
+  if DRStats.circle > 60
+    buy_Taffelberries if 'and see' != DRC.bput("rummage /C Taffelberries my #{$settings.theurgy_supply_container}", 'but there is nothing in there like that', 'and see')
```

Comment:
Did you look into using the `exists?` method (or `inside?`) to replace the rummages?